internal/services/environment/models.go

```diff
@@ -221,6 +222,10 @@ func convertSourceModelFromEnvironmentDto(environmentDto EnvironmentDto, currenc
 		attrValuesProductProperties["language_code"] = types.Int64Value(int64(environmentDto.Properties.LinkedEnvironmentMetadata.BaseLanguage))
 		attrValuesProductProperties["version"] = types.StringValue(environmentDto.Properties.LinkedEnvironmentMetadata.Version)
 		attrValuesProductProperties["unique_name"] = types.StringValue(environmentDto.Properties.LinkedEnvironmentMetadata.UniqueName)
+		if environmentDto.Properties.EnvironmentSku == "Developer" {
+			attrValuesProductProperties["security_group_id"] = types.StringNull()
+			model.OwnerId = types.StringPointerValue(ownerId)
+		}
```

Comment:
consider a validator if these settings are mutually exclusive
https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators/objectvalidator#ConflictsWith

Reply:
I tried but it does not work in that use case. Furthermore, in this code piece case, I have to set sec group as null because for dev environments I cannot be send back over api, otherwise even is its send but "", api  will complain